src/IB/IBFESurfaceMethod.cpp

```diff
@@ -1388,162 +2289,1758 @@ void IBFESurfaceMethod::endDataRedistribution(Pointer<PatchHierarchy<NDIM> > /*h
+
+                // Setup a unit vector pointing in the coordinate direction of
+                // interest.
+                VectorValue<double> q;
+                q(axis) = 1.0;
+
+                // Loop over the relevant range of indices.
+                Box<NDIM> axis_box = box;
+                axis_box.lower(axis) = 0;
+                axis_box.upper(axis) = 0;
+
+                unsigned int SideDim[NDIM][NDIM - 1];
+                for (unsigned int d = 0; d < NDIM; ++d)
+                    for (unsigned int l = 0; l < NDIM - 1; ++l) SideDim[d][l] = (d + l + 1) % NDIM;
+
+                for (BoxIterator<NDIM> b(axis_box); b; b++)
+                {
+                    const Index<NDIM>& i_c = b();
+                    libMesh::Point r;
+                    boost::array<libMesh::Point, NDIM - 1> rs;
+
+                    for (unsigned int d = 0; d < NDIM; ++d)
+                    {
+                        r(d) = (d == axis ? 0.0 :
+                                            x_lower[d] + dx[d] * (static_cast<double>(i_c(d) - patch_lower[d]) + 0.5));
+
+                        for (unsigned int l = 0; l < NDIM - 1; ++l)
+                        {
+                            rs[l](d) =
+                                (d == axis ? 0.0 :
+                                             d == SideDim[axis][l] ?
+                                             x_lower[d] + dx[d] * (static_cast<double>(i_c(d) - patch_lower[d])) :
+                                             x_lower[d] + dx[d] * (static_cast<double>(i_c(d) - patch_lower[d]) +
+                                                                   0.5)); // In 2D this
+                        }
+                    }
+
+                    std::vector<std::pair<double, libMesh::Point> > intersections;
+                    boost::array<std::vector<std::pair<double, libMesh::Point> >, NDIM - 1> intersectionsSide;
+
+                    static const double tolerance = sqrt(std::numeric_limits<double>::epsilon());
+
+#if (NDIM == 2)
+                    intersect_line_with_edge(intersections, static_cast<Edge*>(elem), r, q, tolerance);
+#endif
+#if (NDIM == 3)
+                    intersect_line_with_face(intersections, static_cast<Face*>(elem), r, q, tolerance);
+#endif
+                    for (unsigned int l = 0; l < NDIM - 1; ++l)
+                    {
+#if (NDIM == 2)
+                        intersect_line_with_edge(intersectionsSide[l], static_cast<Edge*>(elem), rs[l], q, tolerance);
+#endif
+#if (NDIM == 3)
+                        intersect_line_with_face(intersectionsSide[l], static_cast<Face*>(elem), rs[l], q, tolerance);
+#endif
+                    }
+
+                    if (d_use_pressure_jump_conditions)
+                    {
+                        for (unsigned int k = 0; k < intersections.size(); ++k)
+                        {
+                            const libMesh::Point x = r + intersections[k].first * q;
```

Comment:
I think we could move this intersection calculation into an exterior function.

Reply:
I might actually decide to remove imposePointWiseJumpConditions all together. It was just for comparing with the weak version of the jump quantities. @boyceg What do you think? 

Also I don't think anything has changed in imposeWeakJumpConditions since the last merge into master, but we can still reduce the number lines or create separate functions like the one you've mentioned here.

Reply:
I would remove any un-needed code.

I also would move intersection related code into helper functions, because we will be wanting to find intersections in other places, too.

Reply:
A good rule of thumb (not always followed by me!) is: if you have a nontrivial piece of code and use it more than once, put it in a helper function.

Reply:
well, if it works, I see no reason to get rid of it.

I would still like to move the intersection algorithm out since its probably useful in other parts of IBAMR.